File: javascript/robot-name/robot-name.js

var usedNames = {}

var Robot = function() {
    this.chars="ABCDEFGHIJKLMNOPQRSTUVWXYZ"
    this.reset()
}

Robot.prototype.reset = function() {
    do {
        this.name =
                this.chars[Math.floor(Math.random() * 26)]
                + this.chars[Math.floor(Math.random() * 26)]
                + Math.floor(Math.random() * 10)
                + Math.floor(Math.random() * 10)
                + Math.floor(Math.random() * 10)
    } while(usedNames[this.name] == true)

    usedNames[this.name] = true
}

module.exports = Robot

File: python/robot-name/robot_name.py
import random
import os

asciiA = ord('A')
asciiZ = ord('Z')

def generateRobotName():
    random.seed(os.urandom(10))
    return generateRandomLetter() \
            + generateRandomLetter() \
            + generateRandom3DigitNumber()

def generateRandom3DigitNumber():
    randomNumber = random.randint(0,999)
    return "{:03d}".format(randomNumber)

def generateRandomLetter():
    charCode = random.randint(asciiA, asciiZ)
    return chr(charCode)

class Robot:
    def __init__(self):
        self.name = generateRobotName()

    def reset(self):
        self.name = generateRobotName()

